### PR TITLE
feat: support serverScope

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

scope is a deploy unit, you can set different config between scopes.

Example:

Deploy application in us and asia with different config，you can set

- config.us_prod.js
- config.ap_prod.js

### CODE REVIEW

**Comment by popomore** [comment]
这个功能暂时不对外开放，文档先不写

File: lib/loader/mixin/config.js [hunk 1]
```diff
@@ -23,10 +23,20 @@ module.exports = {
...
+    let names;
+    if (this.serverEnv === 'default') {
+      names = [
+        'config.default.js',
+        `config.${this.serverScope}.js`,
+      ];
+    } else {
+      names = [
+        'config.default.js',
+        `config.${this.serverScope}.js`,
```
Comment (fengmk2) [inline lib/loader/mixin/config.js]: serverScope 为空不要加载吧

**Comment by atian25** [comment]
哈，记得之前 UC 国际那边也是希望有这个。

`config.${scope}_$env}.js` 又多了一种命名规范啊

**Comment by fengmk2** [comment]
或者命名规范改为 `config.${scope}.${env}.js` ?

**Comment by popomore** [comment]
感觉不要用 . 区分，us_prod 就认为是一个类型就好了。

**Comment by fengmk2** [comment]
@popomore 有道理， @atian25 再看看是否有问题。

**Comment by fengmk2** [comment]
3.15.0

**Pull request merged**